src/c++/perf_analyzer/load_manager.cc

```diff
@@ -110,6 +110,40 @@ LoadManager::GetAccumulatedClientStat(cb::InferStat* contexts_stat)
   return cb::Error::Success;
 }
 
+uint64_t
+LoadManager::GetIdleTime()
+{
+  uint64_t total{0};
+  size_t num_active_threads = 0;
+  for (auto& thread_stat : threads_stat_) {
+    std::lock_guard<std::mutex> lock(thread_stat->mu_);
+    thread_stat->idle_timer.Restart();
+    uint64_t idle_time = thread_stat->idle_timer.GetIdleTime();
+    if (idle_time) {
+      total += idle_time;
+      num_active_threads++;
+    }
+  }
+
+  // TODO REFACTOR TMA-1043 InferDataManager should have an API to get
+  // num_active_threads. This method of determining active threads isn't fully
+  // accurate
```

Comment:
Can you explain why it isn't fully accurate?

Reply:
This code is guessing how many active threads there are by counting how many have non-zero idle time. While this will be correct 99% of the time, it is possible for a thread to have no idle time (if it somehow takes a whole window to prep a request). A more accurate way is to explicitly track the number of active threads, which is something that story is already going to accomplish.